src/main/java/tw/commonground/backend/service/fact/FactService.java

```diff
@@ -0,0 +1,173 @@
+package tw.commonground.backend.service.fact;
+
+import jakarta.servlet.http.HttpServletRequest;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Pageable;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+import tw.commonground.backend.service.fact.dao.FactMapper;
+import tw.commonground.backend.service.fact.dao.FactRequest;
+import tw.commonground.backend.service.fact.dao.FactResponse;
+import tw.commonground.backend.service.fact.entity.FactEntity;
+import tw.commonground.backend.service.fact.entity.FactRepository;
+import tw.commonground.backend.service.reference.*;
+import tw.commonground.backend.shared.exceptions.ExceptionResponse;
+import tw.commonground.backend.shared.exceptions.IdNotFoundException;
+import tw.commonground.backend.shared.pagination.WrappedPaginationResponse;
+import tw.commonground.backend.shared.pagination.PaginationResponse;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Service
+public class FactService {
+
+    private final FactRepository factRepository;
+
+    private final ReferenceRepository referenceRepository;
+
+    private final FactMapper factMapper = new FactMapper();
+    private final ReferenceMapper referenceMapper = new ReferenceMapper();
+
+    public FactService(FactRepository factRepository, ReferenceRepository referenceRepository) {
+        this.factRepository = factRepository;
+        this.referenceRepository = referenceRepository;
+    }
+
+    public WrappedPaginationResponse<List<FactResponse>> getFacts(int page, int size, String column, String mode) {
+
+        Sort sorted = Sort.by((Objects.equals(mode, "desc") ? Sort.Order.desc(column) : Sort.Order.asc(column)));
+        Pageable pageable = PageRequest.of(page, size, sorted);
+        Page<FactEntity> pageFacts = factRepository.findAll(pageable);
+
+        List<FactResponse> factResponses = pageFacts.getContent()
+                .stream()
+                .map(factMapper::toResponse)
+                .toList();
+
+        return new WrappedPaginationResponse<>(factResponses, PaginationResponse.builder()
```

Comment:
The conversion from Entity to DTO should be the responsibility of the Controller,the Service layer should not have any knowledge of DTOs.